### PR TITLE
Add support for OpenAI Assistants

### PR DESCRIPTION
**Pull request by homanp**
## Summary

Add support for running openai assistants.

## Checklist

- [x] My code follows the code style of this project and passes `./lint-and-format.sh format`.
- [x] My change requires a change to the documentation.
- [ ] I have [updated the documentation](https://docs.superagent.sh/) accordingly.
- [ ] My change has adequate unit test coverage.

### CODE REVIEW

File: libs/superagent/app/models/request.py [hunk 1]
```diff
@@ -12,12 +12,14 @@ class ApiUser(BaseModel):
 class Agent(BaseModel):
     isActive: bool = True
     name: str
+    type: Optional[str] = "SUPERAGENT"
     initialMessage: Optional[str]
     prompt: Optional[str]
     llmModel: Optional[str]
-    llmProvider: Optional[LLMProvider]  # either llmProvider or llmModel must be set
-    description: Optional[str] = "An helpful agent."
```
Comment (elisalimli) [inline libs/superagent/app/models/request.py:19]: Why did you remove this? In the database schema the description field is not optional.

Comment (homanp) [reply libs/superagent/app/models/request.py:19]: Correct, I forgot the migrations
